pshops.js

```diff
@@ -0,0 +1,91 @@
+/*
+OAF RW PShops Client Portal: allows clients to activate, top up, and check balance for solar products.
+*/
+
+//global functions -- ripped from core.js, may need to modify
+var msgs = require('./lib/msg-retrieve'); //global message handler
+var admin_alert = require('./lib/admin-alert'); //global admin alerter
+var get_menu_option = require('./lib/get-menu-option');
+var populate_menu = require('./lib/populate-menu');
+
+// set various constants using telerivet tables
+var settings_table = project.getOrCreateDataTable('ussd_settings');
+const max_digits_for_account_number = parseInt(settings_table.queryRows({'vars' : {'settings' : 'max_digits_an'}}).next().vars.value);
+const timeout_length = 180; // what unit is this? seconds?
+
+// display welcome message and prompt input. the input received here will trigger a series of input handlers.
+global.main = function() {
+    sayText(msgs('pshops_main_splash'));
+    promptDigits('account_number_splash', { 'submitOnHash' : false,
+                                            'maxDigits'    : max_digits_for_account_number,
+                                            'timeout'      : 180 });
+}
+
+// input handler for account #
+addInputHandler('account_number_splash', function(accnum){
+    try{ // check if account number is a valid p-shop number
+        check_account_no(accnum);
```

Comment:
probably better if this returns a boolean rather than setting a global variable - then you can move the function call to within the conditional check